modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/loader/parser/metadata/MuleSdkArtifactLocalExtensionModelMetadataParser.java

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.mule.internal.loader.parser.metadata;
+
+import static org.mule.runtime.api.meta.Category.COMMUNITY;
+import static org.mule.runtime.internal.dsl.DslConstants.THIS_NAMESPACE;
+import static org.mule.runtime.internal.dsl.DslConstants.THIS_PREFIX;
+import static org.mule.runtime.module.extension.mule.internal.loader.parser.MuleSdkExtensionModelParser.ARTIFACT_LOCAL_EXTENSION_NAMESPACE;
+import static org.mule.sdk.api.annotation.Extension.MULESOFT;
+
+import static java.util.Optional.empty;
+import static java.util.Optional.of;
+
+import org.mule.runtime.api.meta.Category;
+import org.mule.runtime.module.extension.internal.loader.java.property.LicenseModelProperty;
+import org.mule.runtime.module.extension.internal.loader.parser.XmlDslConfiguration;
+
+import java.util.Optional;
+
+/**
+ * {@link MuleSdkExtensionModelMetadataParser} implementation for Mule SDK extensions defined as part of the same artifact.
```

Comment:
this jdco and classname are not clear not clearly reflect the domain. Be clear about this being for operations defined as part of an app

Reply:
Originally I had named it `MuleSdkApplicationExtensionModelMetadataParser`. However, I realized that I also needed to generate an `ExtensionModel` for the `this` namespace when loading an extension from its own artifact.

This is why I ended up reusing the same thing, but I renamed it to MuleSdk**ArtifactLocal**ExtensionModelMetadataParser.

So I think the Javadoc and the classname are reflecting what it is actually being used for: parsing the ExtensionModel for the cases in which the ExtensionModel is going to be referenced from the same artifact (either application or extension) in which it is being defined.

I got the concept from [this constant](https://github.com/mulesoft/mule/blob/7bc5e0580c328297c4e453cfc3f9fb33634b73fd/modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/loader/parser/MuleSdkExtensionModelParser.java#L56) which would now be renamed to `ARTIFACT_LOCAL_EXTENSION_NAMESPACE`.

Do you think it is too convoluted?

Reply:
Renamed `ArtifactLocal` to `This`, for example: `MuleSdkArtifactLoaclExtensionModelMetadataParser` to `MuleSdkThisExtensionModelMetadataParser`.
I couldn't use `Application` because it would also be misleading. But I suppose the extension `this` is a clearer concept for someone working on this module. In the comments I explain that these classes are used for extensions that are consumed from the same artifact in which they are defined.